### PR TITLE
Fix DI registrations for ServiceFabric

### PR DESCRIPTION
**Pull request by ReubenBond**
Fixes issues with the Service Fabric providers. Reported by @Kwal (thanks!)

### CODE REVIEW

File: src/Orleans/Runtime/OutsideRuntimeClient.cs [hunk 1]
```diff
@@ -116,6 +116,7 @@ public OutsideRuntimeClient(ClientConfiguration cfg, bool secondary = false)
             services.AddFromExisting<IInternalGrainFactory, GrainFactory>();
             services.AddFromExisting<IGrainReferenceConverter, GrainFactory>();
             services.AddSingleton<MessageFactory>();
+            services.AddSingleton<Func<string, Logger>>(LogManager.GetLogger);
```
Comment (jdom) [inline src/Orleans/Runtime/OutsideRuntimeClient.cs]: This method was only available to internals, do we want to open it up to the public? I'm fine with it, just highlighting this change

Comment (ReubenBond) [inline src/Orleans/Runtime/OutsideRuntimeClient.cs]: Yeah, I think it's better this way. It's not very discoverable, but this signature makes sense to me unless we want to maybe add some explicit logging provider interface, like `ILogManager`. We eventually may, but this is a conservative option for now, IMO. What do you think?

Comment (jdom) [inline src/Orleans/Runtime/OutsideRuntimeClient.cs]: I think it's OK for now... Clients are probably not using the Olreans logger anyway, and this opens it up for extensions to the runtime such as this.
Yes, I think we will revamp it... IMO we should use the Microsoft.Extensions.Logging package and get rid of our abstractions...

Comment (ReubenBond) [inline src/Orleans/Runtime/OutsideRuntimeClient.cs]: That would be much better - more alignment with ASP.NET

**Pull request merged**